pkg/analysis/builtins.go

```diff
@@ -134,9 +135,44 @@ func LoadBuiltinsFromSource(ctx context.Context, contents []byte, path string) (
 	}, nil
 }
 
-func LoadBuiltinModule(ctx context.Context, dir string) (*Builtins, error) {
+// An FS wrapper that keeps track of its full path through Sub() calls
+type moduleFS struct {
+	fs.FS
+	path string
+}
+
+var _ fs.SubFS = &moduleFS{}
+
+func newModuleFS(f fs.FS, path string) *moduleFS {
+	if f == nil {
+		f = os.DirFS(path)
+	}
+	return &moduleFS{FS: f, path: path}
+}
+
+func (m *moduleFS) Sub(name string) (fs.FS, error) {
+	subfs, err := fs.Sub(m.FS, name)
+	if err != nil {
+		return nil, err
+	}
+	return &moduleFS{FS: subfs, path: filepath.Join(m.path, name)}, nil
+}
```

Comment:
Can we use `fs.WalkDir` to do our recursion? IIUC, it gives us the full path for each file, so we wouldn't need any of this.

Reply:
Took a little rework of the previous code to get an iterative walk of the directory into the symbol tree, but I think it worked out pretty well.